src/python/py_openvino_genai.cpp

```diff
@@ -0,0 +1,147 @@
+        .def(py::init<>())
+        .def_property_readonly("texts", [](const DecodedResults &dr) { return pyutils::handle_utf8((std::vector<std::string>)dr); })
+        .def_readonly("scores", &DecodedResults::scores)
+        .def_readonly("perf_metrics", &DecodedResults::perf_metrics)
+        .def("__str__", [](const DecodedResults &dr) -> py::str {
+            auto valid_utf8_strings = pyutils::handle_utf8((std::vector<std::string>)dr);
+            py::str res;
+            if (valid_utf8_strings.size() == 1)
+                return valid_utf8_strings[0];
+            
+            for (size_t i = 0; i < valid_utf8_strings.size() - 1; i++) {
+                res += py::str(std::to_string(dr.scores[i])) + py::str(": ") + valid_utf8_strings[i] + py::str("\n");
+            }
+            res += py::str(std::to_string(dr.scores.back())) + py::str(": ") + valid_utf8_strings[valid_utf8_strings.size() - 1];
+            return res;
+        });
+
+    py::class_<EncodedResults>(m, "EncodedResults", encoded_results_docstring)
+        .def_readonly("tokens", &EncodedResults::tokens)
+        .def_readonly("scores", &EncodedResults::scores)
+        .def_readonly("perf_metrics", &EncodedResults::perf_metrics);
+
+    py::class_<StreamerBase, ConstructableStreamer, std::shared_ptr<StreamerBase>>(m, "StreamerBase", streamer_base_docstring)  // Change the holder form unique_ptr to shared_ptr
+        .def(py::init<>())
+        .def("put", &StreamerBase::put, "Put is called every time new token is decoded. Returns a bool flag to indicate whether generation should be stoped, if return true generation stops")
+        .def("end", &StreamerBase::end, "End is called at the end of generation. It can be used to flush cache if your own streamer has one");
+
+    py::class_<ov::Any>(m, "draft_model", py::module_local(), "This class is used to enable Speculative Decoding")
+        .def(py::init([](
+            const std::string& model_path,
+            const std::string& device,
+            const py::kwargs& kwargs
+        ) {
+            return ov::genai::_draft_model(model_path, device, pyutils::kwargs_to_any_map(kwargs)).second;
+        }),
+        py::arg("model_path"), "folder with openvino_model.xml and openvino_tokenizer[detokenizer].xml files",
+        py::arg("device") = "", "device on which inference will be performed"
+        );
+
+    // init tokenizers
+    init_tokenizer(m);
+
+    // init perf metrics
+    init_perf_metrics(m);
+
+    // init generation config
+    init_generation_config(m);
+
+    // init lora adapters
+    init_lora_adapter(m);
+
+    // init continuous batching pipeline
+    init_continuous_batching_pipeline(m);
+
+    // init LLM pipeline
+    init_llm_pipeline(m);
+
+    // init text2image pipeline
+    init_text2image_pipeline(m);
+
+    // init vlm pipeline
+    init_vlm_pipeline(m);
+
+    // init whisper pipeline
+    init_whisper_pipeline(m);
```

Comment:
The comments following the the function names exactly look like mocking :)

Reply:
agree, dropped comments